common/achievement/helper.ts

```diff
@@ -0,0 +1,88 @@
+import { Prisma } from '@prisma/client';
+import { AchievementContextType } from './types';
+import { achievement_state } from '../../graphql/types/achievement';
+import { Context } from '../notification/types';
+import { User } from '../user';
+import { prisma } from '../prisma';
+import { Achievement_template, User_achievement } from '../../graphql/generated';
+import { renderTemplate } from '../../utils/helpers';
+
+export function transformPrismaJson(json: Prisma.JsonValue): AchievementContextType | null {
+    if (!json['match'] && !json['subcourse']) {
+        return null;
+    }
+    const transformedJson: AchievementContextType = {
+        type: json['match'] ? 'match' : 'subcourse',
+        match: json['match'] ? json['match'] : undefined,
+        subcourse: json['subcourse'] ? json['subcourse'] : undefined,
+    };
+    return transformedJson;
+}
+
+export async function getAchievementContext(user: User, userAchievementContext: AchievementContextType): Promise<Partial<Context>> {
+    const achievementContext: Partial<Context> = {
+        user: { ...user, fullName: `${user.firstname} ${user.lastname}` },
+    };
+
+    if (userAchievementContext) {
+        const contextKeys = Object.keys(userAchievementContext);
+        const newContextValue = await Promise.all(
+            contextKeys.map(async (key) => {
+                const [type, id] = userAchievementContext[key].split('/');
+                const newContextValue = await prisma[type].findUnique({
+                    where: { id: Number(id) },
+                });
+                return { key: type, value: newContextValue };
+            })
+        );
+        newContextValue.forEach((context) => {
+            achievementContext[context.key] = context.value;
+        });
+    }
+
+    return achievementContext;
+}
+
+export function getCurrentAchievementContext(userAchievement: User_achievement, achievementContext: Partial<Context>): Achievement_template {
```

Comment:
The name of the function is wrong. It's not returning a context, but a rendered template.